src/sage/algebras/lie_algebras/superliealgebra.py

```diff
@@ -0,0 +1,354 @@
+                           for k in s_coeff}
+            except (KeyError, ValueError):
+                pass
+
+        s_coeff = SuperLieAlgebra._standardize_s_coeff(names, s_coeff, index_set, degrees)
+
+        return super().__classcall__(cls, base=base, names=names, s_coeff=s_coeff, index_set=index_set,
+                                     degrees=degrees, category=category, **kwargs)
+
+    #Standardizing the bracket dictionary:
+
+    def _standardize_s_coeff(names, s_coeff, index_set, degrees):
+
+        index_to_pos = {k: i for i,k in enumerate(index_set)}
+
+        sc = {}
+
+        for k in s_coeff.keys():
+            v = s_coeff[k]
+            if isinstance(v, dict):
+                v = v.items()
+
+            if index_to_pos[k[0]] > index_to_pos[k[1]]:
+                li=names.index(k[0])
+                lj=names.index(k[1])
+                i = degrees[li]
+                j=degrees[lj]
+                key = (k[1], k[0])
+                vals = tuple((g, (-1)**(i*j+1)*val) for g, val in v if val != 0)
+            else:
+                key = tuple(k)
+                vals = tuple((g, val) for g, val in v if val != 0)
+
+            if key in sc.keys() and sorted(sc[key]) != sorted(vals):
+                raise ValueError("two distinct values given for one and the same bracket")
+
+            if vals:
+                sc[key] = vals
+        return Family(sc)
+
+    def __init__(self, base, s_coeff, names, index_set, degrees, category=None):
+
+
+        self._index_to_pos = {k: i for i,k in enumerate(index_set)}
+
+        self._names = names
+        self._ngens = len(self._names)
+        self._degrees = degrees
+        self.name_degree_map = dict(zip(names, degrees))
+        self._M = FreeModule(base, len(index_set))
+        self.s_coeff = s_coeff
+        self.vector_presentation = dict(zip(names, self._M.basis()))
+
+        def to_vector(tuples):
+            vec = [base.zero()]*len(index_set)
+            for k,c in tuples:
+                vec[self._index_to_pos[k]] = c
+            vec = self._M(vec)
+            vec.set_immutable()
+            return vec
+
+        self._s_coeff = {(self._index_to_pos[k[0]], self._index_to_pos[k[1]]): to_vector(s_coeff[k]) for k in s_coeff.keys()}
+
+        base_cat = LieAlgebras(base).WithBasis().FiniteDimensional().or_subcategory(category)
+        category = base_cat.or_subcategory(category, join=True)
+
+        CombinatorialFreeModule.__init__(self, base, index_set)
+
+    def some_elements(self):
+
+        return list(self.basis())
+
+    def base_module(self):
+
+        return VectorSpace(QQ, len(list(self.basis())))
+
+    @cached_method
+    def zero(self):
+
+        return self.element_class(self, {})
+
+    def bracket(self, lt, rt):
+
+        return lt.bracket(rt)
+
+
+    #Morphisms of graded Lie algebras:
+
+    def gradedmorphism(self, on_generators, domain, codomain, argument):
```

Comment:
PEP8 says we should name methods as follows.
```suggestion
    def graded_morphism(self, on_generators, domain, codomain, argument):
```